airflow/www/static/js/graph.js

```diff
@@ -364,14 +364,34 @@ function setFocusMap(state) {
 
 const stateIsSet = () => !!Object.keys(stateFocusMap).find((key) => stateFocusMap[key]);
 
+let prevTis;
+
 function handleRefresh() {
   $('#loading-dots').css('display', 'inline-block');
   $.get(getTaskInstanceURL)
     .done(
       (tis) => {
+        // only refresh if the data has changed
+        if (prevTis !== tis) {
         // eslint-disable-next-line no-global-assign
-        taskInstances = JSON.parse(tis);
-        updateNodesStates(taskInstances);
+          taskInstances = JSON.parse(tis);
+          const states = Object.values(taskInstances).map((ti) => ti.state);
+          updateNodesStates(taskInstances);
+
+          // end refresh if all states are final
+          if (!states.some((state) => (
+            state === null
+            || state === 'running'
+            || state === 'scheduled'
+            || state === 'queued'
+            || state === 'up_for_retry'
+            || state === 'up_for_reschedule'
```

Comment:
We should invert this -- there are fewer terminal states (and less chance of us adding any new ones  - we added a new on in https://github.com/apache/airflow/pull/16681 for instance)


So something like

```suggestion
            ["success", "failed", "upstream_failed", "skipped", "removed"].indexOf(state) !== -1
```